### PR TITLE
Implement CreateServer plugin event 

### PR DESCRIPTION
**Pull request by abrl91**
Issue Number: a part of #3854 and #3887

## PR Details
Implement CreateServer plugin event
related pull request form the plugins repository: https://github.com/amplication/plugins/pull/13

## PR Checklist
- [ ] Tests for the changes have been added
- [x] `npm test` doesn't throw any error

### CODE REVIEW

File: packages/amplication-data-service-generator/src/server/create-server.ts [hunk 1]
```diff
@@ -25,22 +18,27 @@ import { createPackageJson } from "./package-json/create-package-json";
...
-  dtos: DTOs,
-  userEntity: Entity,
-  logger: winston.Logger
-): Promise<Module[]> {
-  const { serverDirectories } = DsgContext.getInstance;
+export function createServer(): Promise<Module[]> {
+  return pluginWrapper(createServerInternal, EventNames.CreateServer, {});
+}
 
+async function createServerInternal(): Promise<Module[]> {
```
Comment (yuval-hazaz) [inline packages/amplication-data-service-generator/src/server/create-server.ts:29]:
```suggestion
async function createServerInternal(eventParams: CreateServerParams): Promise<Module[]> {
```

File: packages/amplication-data-service-generator/src/server/create-server.ts [hunk 2]
```diff
@@ -65,8 +63,9 @@ export async function createServerModules(
   logger.info("Creating swagger...");
   const swaggerModule = await createSwagger();
 
+  const userEntity = entities.find((entity) => entity.name === "User");
```
Comment (yuval-hazaz) [inline packages/amplication-data-service-generator/src/server/create-server.ts:66]: import const from packages/amplication-data-service-generator/src/server/user-entity.ts
```suggestion
  const userEntity = entities.find((entity) => entity.name === USER_ENTITY_NAME);
```

File: packages/amplication-data-service-generator/src/server/create-server.ts [hunk 3]
```diff
@@ -65,8 +63,9 @@ export async function createServerModules(
...
   logger.info("Creating seed script...");
-  const seedModule = await createSeedModule(userEntity);
+  const seedModule = await createSeedModule(userEntity as Entity);
```
Comment (yuval-hazaz) [inline packages/amplication-data-service-generator/src/server/create-server.ts:68]: why do we need this?

**Review by yuval-hazaz — CHANGES_REQUESTED** [review]

File: packages/amplication-data-service-generator/src/server/create-server.ts (hunk 3 above)
Comment (abrl91) [reply packages/amplication-data-service-generator/src/server/create-server.ts:68]: because the result from userEntity could be undefined

Comment (abrl91) [reply packages/amplication-data-service-generator/src/server/create-server.ts:68]: image

**Review by yuval-hazaz — APPROVED** [review]

**Pull request merged**